mods/boats/init.lua

```diff
@@ -136,6 +137,14 @@ function boat.on_punch(self, puncher)
 		minetest.after(0.1, function()
 			self.object:remove()
 		end)
+	elseif name ~= self.driver then
+		local driver = minetest.get_player_by_name(self.driver)
+		driver:punch(puncher, time_from_last_punch, tool_capabilities, direction)
+		if driver:get_hp()<=0 then
+			driver:set_detach()
+			player_api.player_attached[driver:get_player_name()] = false
+			self.driver = nil
```

Comment:
Minetest 5.x does this automatically. Death calls `boat.on_detach_child(self, child)`

Reply:
@Arcelmi

Reply:
removed this part but added player_api.player_attached[driver:get_player_name()] = false to boats.on_detach_child, because the animation wasn't reset on death